server/controllers/notification.js

```diff
@@ -0,0 +1,70 @@
+const Notification = require("../models/Notification");
+const User = require("../models/User");
+const asyncHandler = require("express-async-handler");
+
+// @route PUT /profile/edit
+// @desc edit user profile
+// @access Public
+
+exports.createNotification = asyncHandler(async (req, res, next) => {
+const userId = await User.findById(req.user.id);
```

Comment:
Right so this should be changed to just foundUser or something, not the ID

Reply:
I am a little confused. So the reference to our User Model within the Notification document should also be named user then? So when I try to find the notifications that match the userid that is sent in our requests, it is now checking .user against req.user.id, is that correct? I changed the name of the property within the Model and now on the controllers so it is now called user.

Reply:
Sorry about the confusion here, let me clarify some things:
- `User.findById()` method will return you an entire mongoose document, not just its `_id` so it's misleading to call it `userId`.
- When [saving refs](https://mongoosejs.com/docs/populate.html#saving-refs), mongoose expects a valid `ObjectId` from the referenced model (i.e. `User` in this case). Here, we are passing the entire user document instead so I thought this might lead to errors[^1].
- Lastly, it's a matter of preference to call the `ref` field in `Notification` model either `user` or `userId`, but ultimately that field will store just the `_id` of the document and not the entire user document. When we need to, we can [populate](https://mongoosejs.com/docs/populate.html#populate) this field to retrieve the entire user document.

Hope that clears things up 🙏🏼 

[^1]: **NOTE:** I couldn't test this myself but I have not seen this usage anywhere on the internet, have you tested these endpoints?

Reply:
Okay I think I understand now, thank you. 
Yes I have tested all the endpoints with postman, they are working as intended.